src/main/compa/controllers/FriendshipController.java

```diff
@@ -1,143 +1,125 @@
 
-import javax.management.Query;
 import java.util.List;
 
 public class FriendshipController extends Controller {
-    private static final String PREFIX = "/friend";
+    private static final String PREFIX = "/friendship";
 
     private FriendshipDAO friendshipDAO;
     private UserDAO userDAO;
 
-    Gson gson = new GsonBuilder().setPrettyPrinting().create();
-    //TODO CHANGE AUTH TOKEN TO THE HEADER AND NOT THE BODY
-
-    public FriendshipController(ServiceManager serviceManager, Router router, ModelManager modelManager) {
-        super(serviceManager, PREFIX, router);
-        this.registerRoute(HttpMethod.POST, "/", this::addFriendship, "application/json");
-        this.registerRoute(HttpMethod.GET, "/getFriend", this::getFriends, "application/json");
+    public FriendshipController(Container container) {
+        super(PREFIX, container);
+        this.registerAuthRoute(HttpMethod.POST, "/request", this::requestFriendship, "application/json");
+        this.registerAuthRoute(HttpMethod.POST, "/response", this::respondToRequest, "application/json");
+        this.registerAuthRoute(HttpMethod.GET, "/friend_list", this::getFriends, "application/json");
+        this.registerAuthRoute(HttpMethod.GET, "/pending", this::getPendingFriendships, "application/json");
 
-        friendshipDAO = (FriendshipDAO) modelManager.getDAO(Friendship.class);
-        userDAO = (UserDAO) modelManager.getDAO(User.class);
+        friendshipDAO = (FriendshipDAO) container.getDAO(Friendship.class);
+        userDAO = (UserDAO) container.getDAO(User.class);
     }
 
     /**
-     * @api {post} /friendship Add a new friendship
-     * @apiName AddFriendship
+     * @api {post} /friendship/request Add a new friendship
+     * @apiName Request Friendship
      * @apiGroup Friendship
      * @apiParam {String} friend_id : the id of the user you want to become friends with
-     * @apiParam {String} token : your auth token
      * @apiUse FriendshipAlreadyExist
      */
-    private void addFriendship(RoutingContext routingContext) {
-        String[] params = {"friend_id", "token"};
+    private void requestFriendship(User me, RoutingContext routingContext) {
+        GsonService gson = (GsonService) this.get(GsonService.class);
 
-        if(!this.checkParams(routingContext, params)){
+        if(!this.checkParams(routingContext, "friend_id")){
             routingContext.response().end("missing parameter"); //TODO FORMAT
             return;
         }
 
-        User me = userDAO.findOne("token", routingContext.request().getParam("token"));
+        userDAO.findById(routingContext.request().getParam("friend_id"), res1 -> {
 
-        if(me == null){
-            routingContext.response().end("can't find authenticated user"); //TODO FORMAT
-            return;
-        }
+            User friend = res1.result();
+
+            if(friend == null){
+                routingContext.response().end("can't find your friend"); //TODO FORMAT
```

Comment:
@ssssarah the formatting of the answer could be done from the `checkParams` method

Reply:
check params only checks whether the request contains the parameters we need. 
Here, we're in a situation where the parameter does exist, but when we use it to find the user instance, we find nothing. (the id is not valid)